rust/core/src/automatic_branching.rs

```diff
@@ -1031,4 +1222,62 @@ sources:
 
         assert!(derive_hash_views(&database, &project, &graph).is_ok());
     }
+
+    #[tokio::test]
+    async fn test_derive_hash_views_includes_snapshots() {
+        let fs = quary_proto::FileSystem {
+            files: HashMap::from([
+                (
+                    "models/orders_snapshot.snapshot.sql".to_string(),
+                    File {
+                        name: "models/orders_snapshot.snapshot.sql".to_string(),
+                        contents: prost::bytes::Bytes::from(
+                            r#"
+                          SELECT * FROM q.raw_orders
+                        "#,
+                        ),
+                    },
+                ),
+                (
+                    "models/schema.yaml".to_string(),
+                    File {
+                        name: "models/schema.yaml".to_string(),
+                        contents: prost::bytes::Bytes::from(
+                            r#"
+                            snapshots:
+                            - name: orders_snapshot
+                              unique_key: id
+                              strategy:
+                                timestamp:
+                                  updated_at: updated_at
+                            sources:
+                            - name: raw_orders
+                              path: project_id.dataset_id.orders_snapshot
+                            "#
+                            .to_string(),
+                        ),
+                    },
+                ),
+                (
+                    "quary.yaml".to_string(),
+                    File {
+                        name: "quary.yaml".to_string(),
+                        contents: prost::bytes::Bytes::from(
+                            r#"
+                        sqliteInMemory: {}
+                        "#
+                            .as_bytes(),
+                        ),
+                    },
+                ),
+            ]),
+        };
+        let database = DatabaseQueryGeneratorSqlite {};
+        let project = parse_project(&fs, &database, "").await.unwrap();
+        let graph = project_to_graph(project.clone()).unwrap();
+
+        let hashes = derive_hash_views(&database, &project, &graph).unwrap();
+        
+        assert!(hashes.contains_key("orders_snapshot"));
```

Comment:
Even if not hugely useful, I think it would catch changes if we "hardcoded" the value here of the actual hash it also validates that the interior is something somewhat expected.